src/hotspot/share/cds/filemap.cpp

```diff
@@ -2245,7 +2245,7 @@ address FileMapInfo::heap_region_requested_address() {
     // Runtime base = 0x4000 and shift is also 0. If we map this region at 0x5000, then
     // the value P can remain 0x1200. The decoded address = (0x4000 + (0x1200 << 0)) = 0x5200,
     // which is the runtime location of the referenced object.
-    return /*runtime*/ CompressedOops::base() + r->mapping_offset();
+    return /*runtime*/ (address)((uintptr_t)CompressedOops::base() + r->mapping_offset());
```

Comment:
If `base()` can return 0 (nullptr) then how does the casting help with the warning?

Reply:
I think the addition on uintptr_t is defined in C++ while the addition on nullptr is not.
That's why the ubsan warning/error goes away.

Reply:
Ah I misread the placement of the parentheses.